smart-contracts/contracts/BurnMyWallet.sol

```diff
@@ -26,13 +29,12 @@ contract BurnMyWallet is ERC721, Ownable {
     mint a BURN token to the sender
     */
     function mint() public {
-        require(
-            address(this).balance == 0,
-            "Err: only one BURN token may be minted per account"
-        );
-        uint256 tokenId = _tokenIdCounter.current();
-        _tokenIdCounter.increment();
-        _safeMint(msg.sender, tokenId);
+        _burnWallet(msg.sender);
+    }
+
+    function burnWithMessageSignature(address _signer, uint256 _nonce, bytes memory _signature) public {
```

Comment:
Really like this approach.

One Vulnerability this made me think of for both `burnWithMessageSignature` and even our regular `mint` is that a malicious page could make our user Burn themselves on accident.

You could make a Scam NFT mint page that when you try to mint what you think is some NFT art project instead executes the BurnMyWallet mint, making the user accidentally destroy their account.

Need to think more about how to prevent this in general 🤔 

Might have to ask Wallet Apps to show a big warning message when interacting with this contract unless we figure out some way to prevent this On Chain.

One idea: Make Burning a Wallet a 2-step process. So the first is  Lock that locks the wallet down for a week or a day. And if during that lock Period you submit the Full Burn, then it actually gives you the Burn Token.

So it could first give you an expiring Lock token, and after being Locked for a time period, you have a window during which you can Burn. 

This would prevent a webpage tricking you into quickly burning your wallet, since you would only lock it, and then would need to come back and confirm a day later or else it auto unlocks (or could keep locking it if you wanted to keep it locked for a while if you just suspected a hacker were not ready to Burn fully yet

Reply:
Added an issue to track this here https://github.com/NikitaVr/burnmywallet/issues/4